slsim/Sources/Supernovae/supernovae_pop.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import scipy.integrate as integrate
+import scipy.interpolate as interp
+
+"""References:
+Oguri and Marshall 2010
+"""
+
+
+def calculate_star_formation_rate(z):
+    """Calculates the cosmic star formation rate. (Eq 13 - Oguri and Marshall 2010)
+
+    :param z: redshift
+    :return: cosmic star formation rate in [(h)yr^(-1)Mpc^(-3)]
+    """
+    if z >= 0:
```

Comment:
do we need this condition? The reason I am asking is that if not, then this definition can be also executed array-based

Reply:
If it's fine to have negative redshifts allowed as input (which can result in some formation rates written as complex numbers), then I can remove the condition. Generally, is it fine to assume the user knows this or should I put something in the documentation?

Reply:
you can state also in the input argument that z>=0 has to be satisfied. Then when the user choses other inputs, the definition can not guarantee any meaningful output